packages/react/src/components/MenuButton/MenuButton.tsx

```diff
@@ -0,0 +1,297 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  AutoControlledComponent,
+  RenderResultConfig,
+  applyAccessibilityKeyHandlers,
+  getOrGenerateIdFromShorthand,
+  commonPropTypes,
+  StyledComponentProps,
+} from '../../lib'
+import { ShorthandValue, ComponentEventHandler, ShorthandCollection } from '../../types'
+
+import { Accessibility } from '../../lib/accessibility/types'
+import { createShorthandFactory } from '../../lib/factories'
+import Popup, { PopupProps, PopupEvents, PopupEventsArray } from '../Popup/Popup'
+import Menu, { MenuProps } from '../Menu/Menu'
+import { MenuItemProps } from '../Menu/MenuItem'
+import { Ref } from '@stardust-ui/react-component-ref'
+import { menuButtonBehavior } from '../../lib/accessibility'
+import { focusMenuItem, focusNearest } from './focusUtils'
+import { ALIGNMENTS, POSITIONS, PositioningProps } from '../../lib/positioner'
+
+export interface MenuButtonSlotClassNames {
+  menu: string
+}
+
+export interface MenuButtonProps extends StyledComponentProps<MenuButtonProps>, PositioningProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default menuButtonBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Additional CSS class name(s) to apply.  */
+  className?: string
+
+  /** Initial value for 'open'. */
+  defaultOpen?: boolean
+
+  /** Whether the Popup should be rendered inline with the trigger or in the body. */
+  inline?: boolean
+
+  /** Existing document the popup should add listeners. */
+  mountDocument?: Document
+
+  /** Existing element the popup should be bound to. */
+  mountNode?: HTMLElement
+
+  /** Delay in ms for the mouse leave event, before the popup will be closed. */
+  mouseLeaveDelay?: number
+
+  /** Events triggering the popup. */
+  on?: PopupEvents | PopupEventsArray
+
+  /** Defines whether popup is displayed. */
+  open?: boolean
+
+  /**
+   * Event for request to change 'open' value.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onOpenChange?: ComponentEventHandler<PopupProps>
+
+  /** A popup can show a pointer to trigger. */
+  pointing?: boolean
+
+  /**
+   * DOM element that should be used as popup's target - instead of 'trigger' element that is used by default.
+   */
+  target?: HTMLElement
+
+  /** Element to be rendered in-place where the popup is defined. */
+  trigger?: JSX.Element
+
+  /** Whether the trigger should be tabbable */
+  shouldTriggerBeTabbable?: boolean
```

Comment:
We don't follow the boolean prefixing props, like `should*`.  Example, we use `pointing` instead of `isPointing`, etc.

Propose something like `tabbableTrigger` for this instead.  The props are meant to be read together as a natural language string:

"An open tabbable trigger menu button" ==
```jsx
<MenuButton open tabbableTrigger />
```